### ISSUE TITLE
vscode crash and logout on ubuntu when starting devcontainer

### ISSUE THREAD

**Issue by tepene** [microsoft/vscode#118736]
- VS Code Version: 1.54.1
- OS Version: Ubuntu Budgie 20.04

vscde crashes and Ubuntu user session gets terminated (log out) when starting a devcontainer. Devcontainers are working in vscode version 1.53.2. It looks to me that something changed which causes the issue.

Steps to Reproduce:

1. Start VSCode
2. Open project folder
3. Run "Remote-Containers: Reopen in Container"

Does this issue occur when all extensions are disabled?: No, but without remote container extension I can't start the devcontainer

**Comment by tepene** [comment]
> (Experimental duplicate detection)
> Thanks for submitting this issue. Please also check if it is already covered by an existing one, like:
> 
>     * [Ctrl+click on `file:` link in Debug Console in devcontainer opens host filesystem instead (#111143)](https://www.github.com/microsoft/vscode/issues/111143)

No, haven't found a similar issue.

**Comment by tepene** [comment]
I just tested it with code-insiders. Same issue, crash of vscode and direct ubuntu logout. It works with versions < 1.54

**Comment by tepene** [comment]
I did some more testing, since I was using podman to run the containers. I now switched back to docker. But still VSCode crashes and my user session gets terminated. Tried with the latest insider build. No luck either. VSCode version 1.53.2 is working though.

**Comment by tepene** [comment]
strangely enough vscode stops crashing after setting the "crash-report-directory" with `code --crash-reporter-directory ./debug/`

**Comment by tepene** [comment]
Is there a sane way to troubleshoot this issue?

**Comment by tepene** [comment]
Upgraded to 1.55. Still same issue.

**Comment by tepene** [comment]
Insider 1.56. Still same issue.

**Comment by tepene** [comment]
I now tried disabling the gpu with `--disable-gpu`. Seams to work now, but will test further.

**Comment by tepene** [comment]
no luck with -`--disable-gpu` either. Interestingly it worked one time. The next time I launched vscode it crashed again.
I will run further tests with other options disabled.

**Comment by tepene** [comment]
I think I'm getting closer. For some weird reason, at least from my perspective, vscode crashes when I have the "Ubuntu Budgie Global Menu" enabled. <https://github.com/rilian-la-te/vala-panel-appmenu>. It worked with VSCode up until version 1.53.2  and started crashing with version 1.54.

**Comment by chrmarti** [comment]
- Does it only crash with Remote-Containers or also with other features?
- Did it stop crashing for good when you disabled that Global Menu of Ubuntu Budgie?

"crash and logout" sounds like one of the parent processes terminated.

/cc @deepak1556

**Comment by tepene** [comment]
I no longer see crashes with the disabled "Global Menu". I just did a short test without working in remote-containers with the global menu enabled. After re-enabling the global menu it crashes again. Also I noticed a screen flickering with the global menu enabled. It looks like a rendering issue. First I see the VSCode Menu bar in VSCode, then the screen "flickers" and the menu bar gets moved to the top. I don't see that with other programs. Maybe an Electron issue?
